File: flow-api/scripts/fix-boolean-column-def.py
# -*- coding: utf-8 -*-
import re
from pathlib import Path

root = Path(r"f:/Git_Project/yu-flow/yu-flow/flow-api/src/main/java")
pat1 = re.compile(
    r'@Column\(columnDefinition = "tinyint\(1\) default \d+"\)\n(\s*private Boolean )'
)
pat2 = re.compile(
    r'@Column\(name = "([^"]+)", columnDefinition = "tinyint\(1\) default \d+"\)'
)

for f in root.rglob("*DO.java"):
    text = f.read_text(encoding="utf-8")
    n = pat1.sub(r"@Column(nullable = false)\n\1", text)
    n = pat2.sub(r'@Column(name = "\1", nullable = false)', n)
    if n != text:
        f.write_text(n, encoding="utf-8", newline="\n")
        print("updated", f.relative_to(root))

File: flow-api/scripts/fix-boolean-column-def.js
const fs = require('fs');
const path = require('path');

function walk(dir, acc = []) {
  for (const e of fs.readdirSync(dir, { withFileTypes: true })) {
    const p = path.join(dir, e.name);
    if (e.isDirectory()) walk(p, acc);
    else if (e.name.endsWith('DO.java')) acc.push(p);
  }
  return acc;
}

const root = path.join(__dirname, '../src/main/java');
const pat1 =
  /@Column\(columnDefinition = "tinyint\(1\) default \d+"\)\r?\n(\s*private Boolean )/g;
const pat2 =
  /@Column\(name = "([^"]+)", columnDefinition = "tinyint\(1\) default \d+"\)/g;

for (const f of walk(root)) {
  const t = fs.readFileSync(f, 'utf8');
  const n = t
    .replace(pat1, '@Column(nullable = false)\n$1')
    .replace(pat2, '@Column(name = "$1", nullable = false)');
  if (n !== t) {
    fs.writeFileSync(f, n);
    console.log('updated', path.relative(root, f));
  }
}
